### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by JaneDeng28**

### CODE REVIEW

**Review by xianxinzhu — COMMENTED** [review]
Goes well

**Review by patrickkwang — CHANGES_REQUESTED** [review]

**Comment by JaneDeng28** [comment]
I solve conflict with web editor.

File: ehr_analysis.py [hunk 1]
```diff
@@ -1,52 +1,23 @@
...
-def parse_patient(patient: str) -> List[List[str]]: 
-    with open(patient, "r") as file: 
-        rows = file.readlines() 
-        lists = [] 
-    for row in rows: 
-        row = row.strip() 
-        row = row.split("\t") 
-        lists.append(row)
-    return lists 
+def num_older_than(age, patients):
```
Comment (patrickkwang) [inline ehr_analysis.py:7]: Please run `black`.

File: ehr_analysis.py [hunk 2]
```diff
@@ -1,52 +1,23 @@
 from datetime import datetime
-from typing import Union
-from typing import List
 
-"""Data Parsing"""
+def parse_data(filename):
```
Comment (patrickkwang) [inline ehr_analysis.py:3]: Add type hints everywhere.

File: ehr_part3.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+* a Lab class with:
+  * instance attributes for value, units, etc.
+
+'''
+def parse_data(filename):
+    with open(filename) as table:
+        return [line.replace('\n','').split('\t') for line in table][1:]
+
+class patient():
+    global patients, labs
```
Comment (patrickkwang) [inline ehr_part3.py:18]: You shouldn't need any globals.

File: ehr_part3.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  * a [property](https://docs.python.org/3/library/functions.html#property) called `age`
+* a Lab class with:
+  * instance attributes for value, units, etc.
+
+'''
+def parse_data(filename):
+    with open(filename) as table:
+        return [line.replace('\n','').split('\t') for line in table][1:]
+
+class patient():
```
Comment (patrickkwang) [inline ehr_part3.py:17]: Class names should begin with a capital letter.

**Review by patrickkwang — CHANGES_REQUESTED** [review]
We still want to test everything.

File: ehr_part3.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  * instance attributes for gender, DOB, race, etc.
+  * a [property](https://docs.python.org/3/library/functions.html#property) called `age`
+* a Lab class with:
+  * instance attributes for value, units, etc.
+
+"""
+
+class Patient:
+    def __init__(self, PatientID: str) -> None:
+        self.patients = ehr_analysis.parse_patient("test_patient.txt")
```
Comment (patrickkwang) [inline ehr_part3.py:18]: The user of the library should be able to specify their own file. We want to modify the `parse_patient` function to produce these Patient objects.

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: ehr_part3.py (hunk 5 above)
Comment (JaneDeng28) [reply ehr_part3.py]: This issue is solved.

File: ehr_analysis.py [hunk 6]
```diff
@@ -4,51 +4,59 @@
...
-        lists = [] 
-    for row in rows: 
-        row = row.strip() 
-        row = row.split("\t") 
+
+from datetime import datetime
+from typing import List
+
+
+def parse_patient(patient: str) -> List[List[str]]:
```
Comment (patrickkwang) [inline ehr_analysis.py:12]: We want these parsing functions to be of this form:
```suggestion
def parse_patient(filename: str) -> List[Patient]:
```
i.e. we can construct the `Patient` objects from the file data and return them.

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: ehr_analysis.py (hunk 6 above)
Comment (JaneDeng28) [reply ehr_analysis.py]: This issue is solved.

File: ehr_analysis.py [hunk 7]
```diff
@@ -1,54 +1,59 @@
...
 
-def parse_patient(patient: str) -> List[List[str]]: 
-    with open(patient, "r") as file: 
-        rows = file.readlines() 
-        lists = [] 
-    for row in rows: 
-        row = row.strip() 
-        row = row.split("\t") 
+
+def parse_patient(patfilename: str) -> List[Patient]:
```
Comment (patrickkwang) [inline ehr_analysis.py:8]: `mypy` will not like this. Please use `mypy` to validate the type hints. This is the function signature that we want, _and_ we want the function to _actually do this_. Right now it's returning a `list[list[str]]`, but we need to convert each `list[str]` into a `Patient`.

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: ehr_analysis.py (hunk 7 above)
Comment (JaneDeng28) [reply ehr_analysis.py]: this is solved

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: ehr_analysis.py [hunk 8]
```diff
@@ -1,54 +1,45 @@
...
-        rows = file.readlines() 
-        lists = [] 
-    for row in rows: 
-        row = row.strip() 
-        row = row.split("\t") 
-        lists.append(row)
-    return lists 
+def parse_patient(patfilename:List[str]) -> List[Patient]:
+	with open(patfilename) as patient_table:
+		return [line.replace('\n','').split('\t') for line in patient_table][1:]
```
[the remark this answers is not in the document]
Comment (JaneDeng28) [reply ehr_analysis.py]: I resolve this

File: ehr_part3.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        self.race = patfilename[3]
+        self.marital = patfilename[4]
+        self.language = patfilename[5]
+        self.poverty = patfilename[6]
+
+    @property
+    def age(self) -> float:
+        return round((datetime.now() - self.DOB).days / 365, 1)
+
+    def __lt__(self, other: float):
```
Comment (patrickkwang) [inline ehr_part3.py:43]: Where did this come from?

**Review by patrickkwang — APPROVED** [review]

**Pull request merged**